implement a fuller calculation

rjw57 opened the pull request:
This PR implements a fuller calculation of on-costs. It is *highly* recommended that you read the [accompanying documentation](https://43-133829605-gh.circle-artifacts.com/0/doc/index.html) for this PR as it is quite in-depth.

Closes #7.

rjw57 committed:
terminology fix: gross salary -> base salary

The term "gross salary" was used throughout but was incorrect
terminology. "Base salary" better reflects reality.

rjw57 committed:
implement a fuller calculation

This PR implements a fuller calculation of on-costs. It is *highly*
recommended that you read the accompanying documentation for this commit
as it is quite in-depth.

Closes #7.

rjw57 commented:
The documentation has now been built by CircleCI at https://43-133829605-gh.circle-artifacts.com/0/doc/index.html

abrahammartin commented on ucamstaffoncosts/salary/scales.py:
```diff
@@ -0,0 +1,382 @@
+    #: Grade 11
+    GRADE_11 = enum.auto()
+
+    #: Grade 11
```
Typo

abrahammartin commented on ucamstaffoncosts/salary/scales.py:
```diff
@@ -0,0 +1,382 @@
+    #: Grade 11
+    GRADE_12_BAND_1 = enum.auto()
+
+    #: Grade 11
```
Typo

abrahammartin commented on ucamstaffoncosts/salary/scales.py:
```diff
@@ -0,0 +1,382 @@
+    #: Grade 11
+    GRADE_12_BAND_2 = enum.auto()
+
+    #: Grade 11
```
Typo

abrahammartin commented on ucamstaffoncosts/salary/scales.py:
```diff
@@ -0,0 +1,382 @@
+    #: Grade 11
+    GRADE_12_BAND_3 = enum.auto()
+
+    #: Grade 11
```
Typo

abrahammartin commented on ucamstaffoncosts/costs.py:
```diff
@@ -0,0 +1,530 @@
+    #: NHS
+    NHS = enum.auto()
+
+
```
we need to add MRC Pension (fixed at 15.9%)

rjw57 replied:
I'll move this to #6 since I think it makes sense to address it there.

rjw57 committed:
bugfix: stop test depending on today's date

rjw57 committed:
index.rst: refactor test

Move the explanation table printer into its own function for use in
further tests.

rjw57 committed:
add ability to specify contract start date

Separate "from date" (date from which results should be returned) and
"start date" (date which employee started being employed). This allows
for an estimate of total expenditure over the life of the employment.

rjw57 committed:
bugfix: typo in Grade enum field descriptions

rjw57 committed:
add occupancy calculation

The occupancy field is easy to add since it just scales the base salary.
Add it in in order to increase the number of roles we can model.

rjw57 committed:
bugfix: costs_by_tax_year: allow until_date <= initial_date

If the function is called with until_date <= initial_date, the internal
list of salaries is empty. In this case we should loop without yielding
a result.
